### ISSUE TITLE
Understanding Apparent Differences between Provided Test/Train Files and Paper Tables

### ISSUE THREAD

**Issue by tallakahath** [ACEsuit/mace-off#7]
Hi MACE-OFF team! We really appreciate the code and datasets provided with the paper as a major step toward reproducibility, as well as the effort that went into making this data publicly available. Working with the prepared datasets, we’ve noted a few issues we’re hoping you could provide some insight into.

The [paper](https://pubs.acs.org/doi/10.1021/jacs.4c07099) references training and test dataset files [here](https://www.repository.cam.ac.uk/items/d50227cd-194f-4ba4-aeb7-2643a69f025f) in “extended XYZ” format:
* **Train dataset:** `train_large_neut_no_bad_clean.xyz`
* **Test dataset:** `test_large_neut_all.tar.gz`

However, we've noticed several discrepancies between what was uploaded, and how the files are described in the paper. We would appreciate any help you can offer in understanding these differences.

# Counts by subset

Table 1 in the [paper](https://pubs.acs.org/doi/10.1021/jacs.4c07099) provides counts by subset (e.g., "Pubchem" or "QMugs") of geometries in both the train and test datasets. However, the files provided do not match the counts in Table 1, nor do they conform to the filtering process described in the paper. 

In the *test dataset*, we find more data points than noted in the paper:
- PubChem has 209 excess points (expected: 33,884, found: 34,093 points), of which 126 either have an atomic force >15 eV/A, or a force-sum-over-atoms of >0.1 eV/A.
- DES370K Dimers has 12 excess points (expected: 13,896, found: 13,908 points), and *none* of these points fail the force-filter test.

In the *train dataset*, we find too few data points:
- PubChem is missing 803 points (expected: 646,821, found: 646,018)
- DES370K Dimers is missing 1 point (expected: 263,065 found: 263,064)
- Dipeptides is missing 4 points (expected: 19,773, found: 19,769)

The missing points in the *train dataset* sum up to the 808 points the SI mentions excluding on the basis of large errors with respect to the described MACE-OFF23(M) model, which makes me think the counting *here* was done before the filtering. Note that applying any sort of force-based filtering would actually cull another ~3,000 points from the *train dataset*, which would only make the mismatch worse.

However, the 221 *extra* points present in the *test dataset* are mysterious and unexplained, and potentially impact the final statistics being computed. Even applying a force filter like described in the SI as applied to SPICE2 does not remove 221 points.

# SMILES in datasets

In addition, we noticed another disagreement between the datasets described in the paper and the provided datasets. Specifically, the paper states in Section 2.2:

> Test/train splitting was performed at the molecule level, ensuring that conformers of the same molecule do not appear in both train/validation and test sets.

The data points in train and test datasets extracted from SPICE, conveniently, carry their explicit hydrogen, mapped SMILES. If we count (subset, hydrogen-mapped stereo-annotated SMILES) tuples in train and test datasets and compare, we see that *all SMILES* present in the test set are also present in the training set. Simple code to perform this analysis on the provided datasets and corresponding output is provided below.

Observing the data overall, the 95/5 split into train/test present in the files provided appears to have been done on a random per-snapshot, rather than per-molecule basis.

Any help or advice you have would be greatly appreciated.

<details>

  <summary>

# Code

</summary>

```python
from collections import Counter
from pathlib import Path

import numpy as np
from ase.io.extxyz import read_xyz

expected_counts_train = {
    "PubChem": 646821,
    "DES370K Monomers": 16861,
    "DES370K Dimers": 263065,
    "Dipeptides": 19773,
    "Solvated Amino Acids": 948,
    "water": 1597,
    "QMugs": 2748,
}

expected_counts_test = {
    "PubChem": 33884,
    "DES370K Monomers": 889,
    "DES370K Dimers": 13896,
    "Dipeptides": 1025,
    "Solvated Amino Acids": 52,
    "water": 84,
    "QMugs": 144,
}


def main():
    print("Test Set: ")
    lines = [ll for ll in read_xyz(Path.open("test_large_neut_all.xyz"), index=slice(None, None, None))]
    # Need to ditch 209 PubChem, and 12 DES370K
    test_smiles = Counter([(ll.info["config_type"], ll.info.get("smiles", "")) for ll in lines])

    found_test = Counter([ll.info["config_type"] for ll in lines])
    print("\nNaive: ")
    for k, v in expected_counts_test.items():
        if found_test[k] > v:
            print(f"{k}: found {found_test[k]} in dataset, expected {v} from paper, EXTRA {found_test[k] - v}")
        elif found_test[k] < v:
            print(f"{k}: found {found_test[k]} in dataset, expected {v} from paper, MISSING {v - found_test[k]}")

    found_test_with_filter = Counter(
        [
            ll.info["config_type"]
            for ll in lines
            if not (
                (np.linalg.norm(ll.arrays["forces"], axis=1) > 15).any()
                or (np.linalg.norm(ll.arrays["forces"].sum(axis=0)) > 0.1)
            )
        ]
    )
    print("\nWith Filter: ")
    for k, v in expected_counts_test.items():
        if found_test_with_filter[k] > v:
            print(f"{k}: found {found_test_with_filter[k]} in dataset, expected {v} from paper, EXTRA {found_test_with_filter[k] - v}")
        elif found_test_with_filter[k] < v:
            print(f"{k}: found {found_test_with_filter[k]} in dataset, expected {v} from paper, MISSING {v - found_test_with_filter[k]}")

    print("\nTrain Set: ")
    lines = [ll for ll in read_xyz(Path.open("train_large_neut_no_bad_clean.xyz"), index=slice(None, None, None))]
    # Need to ditch 209 PubChem, and 12 DES370K
    train_smiles = Counter([(ll.info["config_type"], ll.info.get("smiles", "")) for ll in lines])

    found_train = Counter([ll.info["config_type"] for ll in lines])
    print("\nNaive: ")
    for k, v in expected_counts_train.items():
        if found_train[k] > v:
            print(f"{k}: found {found_train[k]} in dataset, expected {v} from paper, EXTRA {found_train[k] - v}")
        elif found_train[k] < v:
            print(f"{k}: found {found_train[k]} in dataset, expected {v} from paper, MISSING {v - found_train[k]}")

    found_train_with_filter = Counter(
        [
            ll.info["config_type"]
            for ll in lines
            if not (
                (np.linalg.norm(ll.arrays["forces"], axis=1) > 15).any()
                or (np.linalg.norm(ll.arrays["forces"].sum(axis=0)) > 0.1)
            )
        ]
    )
    print("\nWith Filter: ")
    for k, v in expected_counts_train.items():
        if found_train_with_filter[k] > v:
            print(f"{k}: found {found_train_with_filter[k]} in dataset, expected {v} from paper, EXTRA {found_train_with_filter[k] - v}")
        elif found_train_with_filter[k] < v:
            print(f"{k}: found {found_train_with_filter[k]} in dataset, expected {v} from paper, MISSING {v - found_train_with_filter[k]}")

    print("\nSMILES Unique: ")
    for k in sorted(test_smiles.keys()):
        if train_smiles.get(k, 0) == 0:
            print(f"No overlap for {k}: {test_smiles[k]} in test, 0 in train")


if __name__ == "__main__":
    main()
```

</details>

# Code Output

```bash

Naive:
PubChem: found 34093 in dataset, expected 33884 from paper, EXTRA 209
DES370K Dimers: found 13908 in dataset, expected 13896 from paper, EXTRA 12

With Filter:
PubChem: found 33967 in dataset, expected 33884 from paper, EXTRA 83
DES370K Dimers: found 13908 in dataset, expected 13896 from paper, EXTRA 12

Train Set:

Naive:
PubChem: found 646018 in dataset, expected 646821 from paper, MISSING 803
DES370K Dimers: found 263064 in dataset, expected 263065 from paper, MISSING 1
Dipeptides: found 19769 in dataset, expected 19773 from paper, MISSING 4

With Filter:
PubChem: found 643719 in dataset, expected 646821 from paper, MISSING 3102
DES370K Dimers: found 263031 in dataset, expected 263065 from paper, MISSING 34
Dipeptides: found 19768 in dataset, expected 19773 from paper, MISSING 5

SMILES Unique:

```
[note that nothing was printed after `SMILES Unique`, rather than the return being truncated]
